### PR TITLE
AP_RangeFinder: Add new driver for TOFSense F I2C

### PR DESCRIPTION
**Pull request by rishabsingh3003**

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    // send command to take reading
+    return _dev->transfer(cmd, sizeof(cmd), nullptr, 0);
+}
+
+// read - return last value measured by sensor
+bool AP_RangeFinder_TOFSenseF_I2C::get_reading(uint32_t &reading_mm, uint16_t &signal_strength, uint16_t &status)
+{
+
+    struct PACKED {
+        le32_t distance_mm;
```
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp:104]: can just use uint32_t, we are always little endian

File: libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        uint32_t signal_strength_and_status;
+    } packet;
+
+    // take range reading and read back results
+    const bool ret = _dev->transfer(nullptr, 0, (uint8_t *) &packet, sizeof(packet));
+
+    if (ret) {
+        // combine results into distance
+        reading_mm = le32toh(packet.distance_mm);
```
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp:113]: next line assumes little endian, no need to convert

File: libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        signal_strength = (uint16_t)(packet.signal_strength_and_status >> 16);
+        status = (uint16_t)(packet.signal_strength_and_status);
+
+        // trigger a new reading
+        start_reading();
```
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp:118]: this looks like if transfer fails, then no new start_reading() is done ever again

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp (hunk 3 above)
Comment (tridge) [reply libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp:118]: same bug in maxsonar i2c

**Comment by tridge** [comment]
both nooploops should be in build_options.py

**Review by tridge — APPROVED** [review]

**Review by peterbarker — COMMENTED** [review]
No simulator for this kind of sucks.

**Pull request merged**